crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -680,64 +728,173 @@ impl Inner {
+        constraint.constraint_id = constraint_id.0;
+        self.create_constraint_internal(&constraint)?;
+
+        log::trace!("CONSTRAINT CREATED: {}", constraint.constraint_name);
+
+        Ok(constraint_id)
+    }
+
+    fn create_constraint_internal(&mut self, constraint: &ConstraintSchema) -> super::Result<()> {
+        let insert_table = if let Some(insert_table) = self
+            .tx_state
+            .as_mut()
+            .unwrap()
+            .get_insert_table_mut(&TableId(constraint.table_id))
+        {
+            insert_table
+        } else {
+            let row_type = self.row_type_for_table(TableId(constraint.table_id))?;
+            let schema = self.schema_for_table(TableId(constraint.table_id))?;
+            self.tx_state.as_mut().unwrap().insert_tables.insert(
+                TableId(constraint.table_id),
+                Table {
+                    row_type,
+                    schema,
+                    indexes: HashMap::new(),
+                    rows: BTreeMap::new(),
+                },
+            );
+            self.tx_state
+                .as_mut()
+                .unwrap()
+                .get_insert_table_mut(&TableId(constraint.table_id))
+                .unwrap()
+        };
+
+        insert_table.schema.constraints.push(ConstraintSchema {
+            constraint_id: constraint.constraint_id,
+            constraint_name: constraint.constraint_name.clone(),
+            kind: constraint.kind,
+            table_id: constraint.table_id,
+            columns: constraint.columns.clone(),
+        });
+
+        Ok(())
+    }
+
+    fn drop_constraint(&mut self, constraint_id: ConstraintId) -> super::Result<()> {
+        const ST_CONSTRAINTS_CONSTRAINT_ID_COL: ColId = ColId(StConstraintFields::ConstraintId as u32);
+        let old_seq_row = self
+            .iter_by_col_eq(
+                &ST_CONSTRAINTS_ID,
+                &ST_CONSTRAINTS_CONSTRAINT_ID_COL,
+                &AlgebraicValue::U32(constraint_id.0),
+            )?
+            .last()
+            .unwrap()
+            .data;
+        let old_row_id = RowId(old_seq_row.to_data_key());
+        self.delete(&ST_CONSTRAINTS_ID, &old_row_id)?;
+
+        self.drop_constraint_internal(&constraint_id);
+
+        Ok(())
+    }
+
+    fn drop_constraint_internal(&mut self, constraint_id: &ConstraintId) {
+        if let Some(insert_table) = self
+            .tx_state
+            .as_mut()
+            .unwrap()
+            .get_insert_table_mut(&TableId(constraint_id.0))
+        {
+            insert_table
+                .schema
+                .constraints
+                .retain(|x| x.constraint_id != constraint_id.0);
+        }
+    }
+
+    // NOTE: This must be sync with `Self::schema_for_table`
     fn create_table(&mut self, table_schema: TableDef) -> super::Result<TableId> {
-        let table_name = table_schema.table_name.as_str();
+        let table_name = table_schema.table_name.clone();
```

Comment:
Because later I get:

```bash
error[E0505]: cannot move out of `table_schema` because it is borrowed
   --> crates/core/src/db/datastore/locking_tx_datastore/mod.rs:897:28
```

Reply:
Nevermind I fix it.